src/test01/directives/windowResize.js

```diff
@@ -0,0 +1,23 @@
+export function windowResizeDirective() {
+  angular
+    .module('app')
+    .directive('ngWindowResize', WindowResize);
+
+  WindowResize.$inject = ['$window', '$rootScope'];
+  function WindowResize($window, $rootScope) {
+    return {
+      restrict: 'CA',
+      link: function (scope) {
+        angular.element($window).on('resize', function (e) {
+          scope.$broadcast('window-resize', {
+            'eventTarget': e.target
+          });
+
+          $rootScope.$broadcast('window-resize', {
+            'eventTarget': e.target
+          });
+        });
+      }
+    };
+  }
+};
```

Comment:
This directive is used to detect and broadcast window-resize events so that  buttonResize directive can resize the buttons when the window is resized.

This solution is based off of: http://blog.dniccumdesign.com/force-elements-to-be-the-same-height-with-angular-duh/